plugins/modules/tanzu_mission_control_secret.py

```diff
@@ -0,0 +1,793 @@
+            }
+        }
+    }
+    return {".dockerconfigjson": json.dumps(payload)}
+
+
+def main():
+    module = load_module()
+
+    state = module.params["state"]
+    api_host = module.params["api_host"]
+    api_token = module.params["api_token"]
+    cluster_name = module.params["cluster_name"]
+    cluster_namespace = module.params["cluster_namespace"]
+    management_cluster_name = module.params["management_cluster_name"]
+    provisioner_name = module.params["provisioner_name"]
+    secret_name = module.params["secret_name"]
+    data = module.params["data"]
+    cluster_group = module.params["cluster_group"]
+    secret_type = module.params["secret_type"]
+    export = module.params["export"]
+
+    try:
+        access_token = exchange_token(api_token)
+        if access_token is None:
+            return module.fail_json(msg="Failed to exchange token, check credentials")
+    except HTTPError as e:
+        return module.fail_json(msg=e.read().decode())
+
+    tmc_secret = None
+
+    if cluster_name is not None:
+        tmc_secret = TMCClusterSecret(
+            str(management_cluster_name),
+            str(provisioner_name),
+            cluster_name,
+            cluster_namespace,
+            secret_name,
+            secret_type,
+        )
+    else:
+        tmc_secret = TMCClusterGroupSecret(
+            str(cluster_group), cluster_namespace, secret_name, secret_type
+        )
+
+    if secret_type == "SECRET_TYPE_OPAQUE":
+        payload = data
+    else:
+        payload = create_dockerconfig(
+            module.params["registry_host"],
+            module.params["registry_username"],
+            module.params["registry_password"],
+        )
+
+    secret_status = tmc_secret.exists(api_host, access_token)
+    action = handle_logic(secret_status, state)
+    created, create_msg = False, ""
+
+    try:
+        if action == "do_nothing":
+            return module.exit_json(changed=False)
+
+        if action == "delete":
+            changed, msg = tmc_secret.delete(api_host, access_token)
+            return module.exit_json(changed=changed, msg=msg)
+
+        if action == "force_update":
+            created, create_msg = tmc_secret.update(api_host, access_token, payload)
+        elif action == "create":
+            created, create_msg = tmc_secret.create(api_host, access_token, payload)
+        else:
+            return module.fail_json(msg="Unknown action {}".format(action))
+
+        exported, export_msg = False, ""
+
+        if export and not tmc_secret.is_exported(api_host, access_token):
+            exported, export_msg = tmc_secret.create_export(api_host, access_token)
+
+        if not export and tmc_secret.is_exported(api_host, access_token):
+            exported, export_msg = tmc_secret.delete_export(api_host, access_token)
+
+        final_changed = created or exported
+        final_msg = (
+            create_msg if export_msg == "" else create_msg + " and " + export_msg
```

Comment:
It's better to use `str.format()` or `%` instead of concatenating strings manually:
```suggestion
            create_msg if export_msg == "" else "{} and {}".format(create_msg, export_msg)
```